### PR TITLE
Terrain tzw

### PR DESCRIPTION
**Pull request by tangziwen**

### CODE REVIEW

File: cocos/3d/CCTerrain.cpp [hunk 1]
```diff
@@ -0,0 +1,801 @@
...
+        }
+    }
+    calculateAABB();
+    finish();
+}
+
+cocos2d::Terrain::Chunk::Chunk()
+{
+    _currentLod = 0;
+    left =NULL;
```
Comment (super626) [inline cocos/3d/CCTerrain.cpp]: NULL->nullptr

File: cocos/3d/CCTerrain.cpp [hunk 2]
```diff
@@ -0,0 +1,801 @@
...
+        textImage->initWithImageFile(parameter.detailMaps[0].detailMapSrc);
+        auto texture = new Texture2D();
+        texture->initWithImage(textImage);
+        obj->textures.push_back(texture);
+        obj->init();
+        obj->setAnchorPoint(Vec2(0,0));
+    }else
+    {
+        //alpha map
+        auto textImage = new Image();
```
Comment (super626) [inline cocos/3d/CCTerrain.cpp]: new (std::nothrow)

File: cocos/3d/CCTerrain.cpp [hunk 3]
```diff
@@ -0,0 +1,801 @@
...
+    int gridY = _size.height;
+    int gridX = _size.width;
+
+    int step = int(powf(2.0f, float(_currentLod)));
+    if((left&&left->_currentLod > _currentLod) ||(right&&right->_currentLod > _currentLod)
+            ||(back&&back->_currentLod > _currentLod) || (front && front->_currentLod > _currentLod))
+            //need update indices.
+        {
+         //t-junction inner 
+            _lod[_currentLod].indices.clear();
```
Comment (super626) [inline cocos/3d/CCTerrain.cpp:456]: Can we know it size?
If so, _lod.reserve()

Comment (tangziwen) [inline cocos/3d/CCTerrain.cpp:456]: fixe above

**Pull request merged**